### ISSUE TITLE
Cannot authenticate to storage account in AzureUSGovernment

### ISSUE THREAD

**Issue by lclarkjr** [hashicorp/terraform-provider-azurerm#16685]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.9

### AzureRM Provider Version

3.0.0

### Affected Resource(s)/Data Source(s)

storage account

### Terraform Configuration Files

```hcl
terraform {

  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "~> 3.0.0"
    }
  }

backend "azurerm" {
    resource_group_name  = <resource group>
    storage_account_name = <storage account name>
    container_name       = <container name>
    key                  = "tfstate"
    
    #when using this it tries to use the storage url
    #https://<storageaccountname>.blob.core.windows.net/test?comp=list&prefix=tfstateenv%3A&restype=container
    #instead of
    #https://<storageaccountname>.blob.core.usgovcloudapi.net/test
    #access_key = <access key>
  }
}

provider "azurerm" {
  features {}

  environment = "usgovernment"
  #tried with this param and without
  metadata_host = "management.usgovcloudapi.net"

  #tried with and without this section
  #whith it requests fail to respond, without it i get the cannot find subscription with management.azure.com as the url
  #subscription_id = <subscripton id>
  #tenant_id       = <tenant id>

  #note every attempt to get the provider to use the correct url
  #via config params has failed, still always using management.azure.com
  #instead of management.usgovcloudapi.net
}
```

### Debug Output/Panic Output

```shell
2022-05-06T10:34:57.548-0400 [INFO]  Terraform version: 1.1.9
2022-05-06T10:34:57.548-0400 [INFO]  Go runtime version: go1.17.2
2022-05-06T10:34:57.548-0400 [INFO]  CLI args: []string{"terraform", "init"}
2022-05-06T10:34:57.548-0400 [TRACE] Stdout is a terminal of width 235
2022-05-06T10:34:57.548-0400 [TRACE] Stderr is a terminal of width 235
2022-05-06T10:34:57.548-0400 [TRACE] Stdin is a terminal
2022-05-06T10:34:57.548-0400 [DEBUG] Attempting to open CLI config file: /home/clarkl/.terraformrc
2022-05-06T10:34:57.548-0400 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2022-05-06T10:34:57.549-0400 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2022-05-06T10:34:57.549-0400 [DEBUG] ignoring non-existing provider search directory /home/clarkl/.terraform.d/plugins
2022-05-06T10:34:57.549-0400 [DEBUG] ignoring non-existing provider search directory /home/clarkl/.local/share/terraform/plugins
2022-05-06T10:34:57.549-0400 [DEBUG] ignoring non-existing provider search directory /usr/local/share/terraform/plugins
2022-05-06T10:34:57.549-0400 [DEBUG] ignoring non-existing provider search directory /usr/share/terraform/plugins
2022-05-06T10:34:57.549-0400 [DEBUG] ignoring non-existing provider search directory /var/lib/snapd/desktop/terraform/plugins
2022-05-06T10:34:57.550-0400 [INFO]  CLI command args: []string{"init"}

Initializing the backend...
2022-05-06T10:34:57.569-0400 [TRACE] Meta.Backend: built configuration for "azurerm" backend with hash value 1170074985
2022-05-06T10:34:57.569-0400 [TRACE] Meta.Backend: backend has not previously been initialized in this working directory
2022-05-06T10:34:57.569-0400 [DEBUG] New state was assigned lineage "1a68e8cd-be5f-2b52-4d81-75ee51bd989b"
2022-05-06T10:34:57.569-0400 [TRACE] Meta.Backend: moving from default local state only to "azurerm" backend
2022-05-06T10:34:57.573-0400 [DEBUG] checking for provisioner in "."
2022-05-06T10:34:57.576-0400 [DEBUG] checking for provisioner in "/usr/bin"
2022-05-06T10:34:57.577-0400 [TRACE] backend/local: state manager for workspace "default" will:
 - read initial snapshot from terraform.tfstate
 - write new snapshots to terraform.tfstate
 - create any backup at terraform.tfstate.backup
2022-05-06T10:34:57.577-0400 [TRACE] statemgr.Filesystem: reading initial snapshot from terraform.tfstate
2022-05-06T10:34:57.577-0400 [TRACE] statemgr.Filesystem: snapshot file has nil snapshot, but that's okay
2022-05-06T10:34:57.577-0400 [TRACE] statemgr.Filesystem: read nil snapshot
2022-05-06T10:34:57.577-0400 [TRACE] Meta.Backend: ignoring local "default" workspace because its state is empty
2022-05-06T10:34:57.577-0400 [INFO]  Testing if Service Principal / Client Certificate is applicable for Authentication..
2022-05-06T10:34:57.577-0400 [INFO]  Testing if Multi Tenant Service Principal / Client Secret is applicable for Authentication..
2022-05-06T10:34:57.577-0400 [INFO]  Testing if Service Principal / Client Secret is applicable for Authentication..
2022-05-06T10:34:57.577-0400 [INFO]  Testing if Managed Service Identity is applicable for Authentication..
2022-05-06T10:34:57.577-0400 [INFO]  Testing if Obtaining a Multi-tenant token from the Azure CLI is applicable for Authentication..
2022-05-06T10:34:57.577-0400 [INFO]  Testing if Obtaining a token from the Azure CLI is applicable for Authentication..
2022-05-06T10:34:57.577-0400 [INFO]  Using Obtaining a token from the Azure CLI for Authentication
2022-05-06T10:34:58.065-0400 [INFO]  Getting OAuth config for endpoint https://login.microsoftonline.com/ with  tenant <redacted>
2022-05-06T10:34:58.065-0400 [DEBUG] Obtaining a ADAL / Azure Active Directory Graph token for Resource Manager..
2022-05-06T10:34:58.362-0400 [DEBUG] New state was assigned lineage <redacted>
2022-05-06T10:34:58.362-0400 [DEBUG] Building the Container Client from an Access Token (using user credentials)
2022-05-06T10:34:58.363-0400 [DEBUG] Azure Backend Request:
POST /subscriptions/<subscriptionid>/resourceGroups/<resourcegroup>/providers/Microsoft.Storage/storageAccounts/<storageaccount>/listKeys?api-version=2021-01-01 HTTP/1.1
Host: management.azure.com
User-Agent: Terraform/1.1.9
Content-Length: 0
Accept-Encoding: gzip
2022-05-06T10:34:58.670-0400 [DEBUG] Azure Backend Response for https://management.azure.com/subscriptions/<subscriptionid>/resourceGroups/<resourcegroup>/providers/Microsoft.Storage/storageAccounts/<storageaccount>/listKeys?api-version=2021-01-01:
HTTP/2.0 404 Not Found
Content-Length: 129
Cache-Control: no-cache
Content-Type: application/json; charset=utf-8
Date: Fri, 06 May 2022 14:35:58 GMT
Expires: -1
Pragma: no-cache
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-Content-Type-Options: nosniff
X-Ms-Correlation-Request-Id: <redacted>
X-Ms-Failure-Cause: gateway
X-Ms-Request-Id: <redacted>
X-Ms-Routing-Request-Id: CENTRALUS:20220506T143559Z:<redacted>

{"error":{"code":"SubscriptionNotFound","message":"The subscription '<subscriptionid>' could not be found."}}
╷
│ Error: Failed to get existing workspaces: Error retrieving keys for Storage Account "<storageaccount>": storage.AccountsClient#ListKeys: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="SubscriptionNotFound" Message="The subscription '<subscriptionid>' could not be found."
```

### Expected Behaviour

Expect authentication to the storage account and terraform state stored in specified container.

### Actual Behaviour

Failure to authenticate with the storage account with either a no response or a failure to find the subscription id.

It seems that the setting of the environment to usgovernment is not working 

### Steps to Reproduce

az cloud set --name AzureUSGovernment
az login
terraform init

### Important Factoids

Running in AzureUSGovernment

### References

This issue seems to be similar however was closed as being resolved in v2.0.0 however attempting to use that version also fails.
[](https://github.com/hashicorp/terraform-provider-azurerm/issues/5818)

**Comment by tombuildsstuff** [comment]
@lclarkjr this has been fixed in a more recent version of 3.0 - at the time of writing v3.5.0 is the latest release, so you'll need to upgrade to that version for the fix.

Since this should be fixed by upgrading the version of Terraform being used here I'm going to close this issue for the moment, but if that doesn't work for you then please let us know and we can take another look.

Thanks!

**Comment by lclarkjr** [comment]
i just tried it with v3.5.0 with the same result
